sgt-back-end/utils/validationUtils.js

```diff
@@ -0,0 +1,36 @@
+export class CustomError extends Error {
+  constructor(statusCode, message) {
+    super();
+    this.statusCode = statusCode;
+    this.message = message;
+  }
+}
+
+export function validateId(id) {
+  const isValidId = Number.isInteger(id) && id > 0;
+  if (!isValidId) {
+    throw new CustomError(
+      400,
+      "Please provide a valid query parameter for 'gradeId' (positive integer)"
```

Comment:
Not a huge issue, but I'd be wary of calling path parameters "query parameters", since the path and the query string are two different parts of URLs.